### ISSUE TITLE
Does not comply with freedesktop.org Desktop Entry Specification

### ISSUE THREAD

**Issue by dgmulf** [micheleg/dash-to-dock#1305]
According to the [freedesktop.org Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html) (version 1.5 as of time of writing),

> If multiple files have the same desktop file ID, the first one in the $XDG_DATA_DIRS precedence order is used.

In practice, that means users should be able to override system-wide desktop files in `/usr/share/applications` with files in `~/.local/share/applications`, as long as the latter directory is defined first in the variable. GNOME's built-in application launcher adheres to this spec, but on my system (Manjaro GNOME 20.1), this doesn't appear to be the case for Dash to Dock.

Two users reported this problem in issue #462. The user who submitted the report closed the issue claiming user error, but did not specify what he/she did to fix it.

**Comment by ckotte** [comment]
@dgmulf How do you configure $XDG_DATA_DIRS?

I have a local .desktop file that's not used by Dash to Dock and GNOME's application launcher. (It's used if I start the application via Albert)

These folders are currently configured:
```
/home/<user>/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share
```

If I add this to ```~/.config/environment.d/xfg-directories.conf```..
```
XDG_DATA_DIRS=$HOME/.local/share:$XDG_DATA_DIRS
```

..then I cannot login to GNOME anymore:
```
gnome-session-binary[2348]: GLib-GIO-ERROR: No GSettings schemas are installed on the system aborting...
```

**Comment by dgmulf** [comment]
I simply added `~/.local/share` to `~/.profile`, where I define all of my user session environment variables. I haven't played around with systemd's environment functionality.

**Comment by ckotte** [comment]
Ok. This doesn't work with GNOME on Wayland; ```~/.profile``` is ignored.

However, if I use GNOME on Xorg and add ```~/.local/share``` to ```$XDG_DATA_DIRS``` nothing changes. All my local .desktop files are not used by Dash to Dock or GNOME's application launcher..

**Comment by dgmulf** [comment]
Which application are you attempting to launch? I didn't even need to alter `XDG_DATA_DIRS` for the GNOME application launcher to honor my override; it just recognized it out of the box.

**Comment by ckotte** [comment]
I tried Tilix, Chromium, and Vivaldi desktop files. I changed Exec to gnome-terminal (to be absolutely sure) and it only opens gnome-terminal when I start it via Albert.

Maybe there's some configuration missing. I use Arch Linux w/ GNOME 3.36.x.

P.S.: I want to start Tilix maximized and I remove /usr/bin from the browser desktop files because I use firejail, but it looks like it's not necessary (anymore) to use local desktop files for firejail.
